endgame_postprocessing/model_wrappers/sch/run_sch.py

```diff
@@ -47,19 +48,49 @@ def probability_any_worm(probability_for_each_worm: Iterable[float]):
     prob_not_any_worm = reduce(mul, prob_of_not_each_worm, 1.0)
     return 1.0 - prob_not_any_worm
 
+def probability_any_worm_max(probability_for_each_worm: Iterable[float]):
+    """
+    Calculate the probability of having any worm, given by the highest probability
+    among all the worms. Used for SCH.
 
-def canoncialise_single_result(file_info):
-    raw_iu = pd.read_csv(file_info.file_path)
-    raw_without_columns = raw_iu.drop(columns=["intensity", "species"])
-    # TODO: canonical shouldn't need the age_start / age_end but these are assumed present later
-    return canonicalise.canonicalise_raw(
-        raw_without_columns, file_info, "Prevalence SAC"
-    )
+    Inputs:
+     - probability_for_each_worm: Probability of having each worm
 
+    Returns: the probability of having any worm.
+    """
+    return reduce(lambda x, y: np.maximum(x, y), probability_for_each_worm)
+
+def canoncialise_single_result(file_info, warning_if_no_file=False):
+    try:
+        raw_iu = pd.read_csv(file_info.file_path)
+        raw_without_columns = raw_iu.drop(columns=["intensity", "species"])
+        # TODO: canonical shouldn't need the age_start / age_end but these are assumed present later
+        return canonicalise.canonicalise_raw(
+            raw_without_columns, file_info, "Prevalence SAC"
+        )
+    except FileNotFoundError:
+        if warning_if_no_file:
+            warnings.warn(
+                f"File {file_info.file_path} not found, and `warning_if_no_file` is set to False"
+            )
+            return pd.DataFrame()
+        raise FileNotFoundError
+
+
+def combine_many_worms(first_worm, other_worms, combination_function = probability_any_worm):
+    if not callable(combination_function):
+        raise Exception("Need to provide a callable function to combine worms.")
```

Comment:
You could move this test down closer to where `combination_function` is actually used. In addition it is `TypeError` that is typically raised by the Python interpreter when trying to use `__call__` on an object that is not callable.

Reply:
I changed the error type, however I didn't move the check because I prefer raising any exceptions as early in the code as possible. I will be happy to change it if you feel strongly it should be lower (i.e near line 93)